dev/NumberBox/NumberBox.xaml

```diff
@@ -20,6 +20,14 @@
                 <ControlTemplate TargetType="local:NumberBox">
                     <Grid>
                         <VisualStateManager.VisualStateGroups>
+                            <VisualStateGroup x:Name="DisabledStates">
```

Comment:
Convention is to name this group `<VisualStateGroup x:Name="CommonStates">`

Reply:
I followed the NavigationViewItemPresenter style here: https://github.com/microsoft/microsoft-ui-xaml/blob/7acb9a7c5769af3e0b5dbe37c009dcafebef939c/dev/NavigationView/NavigationView_rs1_themeresources.xaml#L605

The reason I did not went with "Common" here is that all the other visual states which are typically part of the "Common" state group like *PointerOver* and *Pressed* are not relevant here (they are already handled by the TextBox control, etc...). As I only have two states I'm interested here - the *Enabled* and *Disabled* states - I felt it would be more suitable to name their parent visual state group accordingly.

As for the naming convention you are mentioning: Observe that the NavigationViewItemPresenter styles use visual state groups like *PointerStates* and *DisabledStates* instead of *Common*.

It is true that many default control templates in UWP overwhelmingly have the *Disabled* visual state as part of the *Common* visual state group and I'm open to change this accordingly. Let's also see what the team thinks.

Reply:
Makes sense. I would have used CommonStates for both controls if it was me but that's just preference. Will agree with the team either way.

Reply:
The big thing to think about with different visual state groups is that states that are not in the same group cannot change the same properties.  For instance if the disabled state changed the background color and the pointer over state also changed the background color, this would result in broken state manager.  This is because there is no mechanism to merge two states (disabled pointerover) and the result is whichever state is set last wins.  I think this is generally the reason the disabled state is included in the common states.

Reply:
So we would be fine here leaving this as it is right now. If changes are made to the default NumberBox control template in the future which will create the above described situation, we can always make the necessary adjustments. 

How does that sound? @StephenLPeters

Reply:
Similar to named template parts or resources, removing or deleting a visual state is a potentially breaking change. I think it would not result in a crash though, calling GoToState("NotAState") just returns false.  We should treat this as we treat template parts and resources, with the goal of hopefully not changing it later.

Reply:
For now, the rest of the NumberBox UI to be updated when then NumberBox is set to disabled is handled by the internal parts of the NumberBox: The TextBox and RepeatButtons used all take care of that themselves. Similary, the PointerOver/Pressed states are also handled by those internal sub parts of the NumberBox. If we were to change that in the future so that the NumberBox itself will take care of updating the properties such as Background, Foreground and BorderBrush for the TextBox and RepeatButtons, then we will likely introduce a `Common` VisualStateGroup and the typical states alongside it - including a `Disabled` visual state. That state would then naturally be the best candidate to update the Header foreground as well - instead of having two disabled states in two different visual state groups with each just settings one distinct subset of the properties which need to be updated when NumberBix is set to disabled.

Ok, long story short, even if the current setup looks fine and won't break in the future there's a chance we might end up with two distinct "Disabled" visual states (in different visual state groups) here in the future. To avoid that possibility outright, we can introduce a `Common` VisualStateGroup with two states for now - `Normal` and `Disabled`. The current "DisabledStates" visual state group will be removed.

That way, I won't be blamed later if we'll end changing the way how the NumberBox UI is updated when the NumberBox is set to disabled 😆

Reply:
> The big thing to think about with different visual state groups is that states that are not in the same group cannot change the same properties. For instance if the disabled state changed the background color and the pointer over state also changed the background color, this would result in broken state manager. This is because there is no mechanism to merge two states (disabled pointerover) and the result is whichever state is set last wins. 

That's interesting and something I never considered before. However, I'm not sure its really as bad as a broken stake manager sounds. Applying the visual state (GoToState) is analogous to just running that state's code and setting the properties as defined. It's perfectly fine if you then apply another state that sets the same properties. After all the states, state groups and even the state manager itself has no higher-level understanding of what it's doing. The logic in ensuring that what states are applied makes sense is handled by the control itself. It's perfectly fine to collapse the same control in two different state/stategroups. That can make sense in some situations and runs perfectly fine in my experience -- nothing breaks.

Reply:
> > The big thing to think about with different visual state groups is that states that are not in the same group cannot change the same properties. For instance if the disabled state changed the background color and the pointer over state also changed the background color, this would result in broken state manager. This is because there is no mechanism to merge two states (disabled pointerover) and the result is whichever state is set last wins.
> 
> That's interesting and something I never considered before. However, I'm not sure its really as bad as a broken stake manager sounds. Applying the visual state (GoToState) is analogous to just running that state's code and setting the properties as defined. It's perfectly fine if you then apply another state that sets the same properties. After all the states, state groups and even the state manager itself has no higher-level understanding of what it's doing. The logic in ensuring that what states are applied makes sense is handled by the control itself. It's perfectly fine to collapse the same control in two different state/stategroups. That can make sense in some situations and runs perfectly fine in my experience -- nothing breaks.

The problem is more pronounced when both states are setting color. Consider if you had selection state seperate from pointer states. The selected state might set the background color to the accent color, while the pointer over state might set it to a light grey.  The result is that the background color ends up being whichever state is entered last. In general we like to limit the amount that order of operations matters.

Reply:
The IsEnabled property is on the Control type, so I think the Control type should already be listening to property changed on that type and sending the control to the CommonStates.Disabled visual state. I think it makes sense to use the CommonStates group and populate Normal and Disabled states there. (and we don't need to duplicate the disabled state here as well as listen to the property change and send to that state). I am not seeing a downside to doing that. 

NavigationViewItemPresenter should probably have followed the same pattern, that is probably an oversight - or perhaps there were other motivations for that case.

Reply:
@ranjeshj While that sounds reasonable I tested removing the IsEnabled property change listener and just use the CommonStates VSG here but then the NumberBox header is not updated at all 🤔I need to explicitly go to the CommonStates.Disabled state here to update the header properly.

Reply:
Interesting. I just looked through the control code and it does not send it to the Disabled state based on IsEnabled. @MikeHillberg Is the expectation that each deriving control do this ?

Reply:
Yes, the visual state names are defined by each control, so the Control class can't know what state names to go to.